### PR TITLE
Optimization process of finding DVS for port

### PR DESCRIPTION
**Pull request by ekosareva**

### CODE REVIEW

File: vmware_dvs/agent/firewalls/vcenter_firewall.py [hunk 1]
```diff
@@ -92,24 +87,29 @@ def security_group_updated(self, action_type, sec_group_ids,
                                device_id=None):
         pass
 
-    def _get_dvs_for_port_id(self, port_id, p_key=None):
-        # Check if port is already known
+    def _update_dvs_port_map(self, ports):
```
Comment (VTabolin) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py:90]: You have only one call from get_dvs_for_port_id with only one port as argument. No need to use list object ports except port.

File: vmware_dvs/utils/dvs_util.py [hunk 2]
```diff
@@ -512,13 +512,14 @@ def create_port_map(dvs_list):
...
 
-def get_dvs_by_id_and_key(dvs_list, port_id, port_key):
+def get_dvs_by_network(dvs_list, network_id):
     for dvs in dvs_list:
-        port = dvs._get_port_info_by_portkey(port_key)
-        if port:
-            if port.config.name == port_id:
+        try:
+            network_name = dvs._get_net_name(dvs.dvs_name, {'id': network_id})
+            if dvs._get_pg_by_name(network_name):
```
Comment (VTabolin) [inline vmware_dvs/utils/dvs_util.py:519]: You does not need PG object itself. In multiple VM creation cases it cause many identical calls to vCenter. I hope that you can improve it by cashing of network names for each DVS and try to find dvs in cash at first.
Do dvs_list cycle on failure in cash only.

File: vmware_dvs/agent/firewalls/vcenter_firewall.py (hunk 1 above)
Comment (ekosareva) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py:90]: No, this functions calls from 2 places: in _apply_sg_rules_for_port function(line 75) and in get_dvs_for_port_id(line 62)

File: vmware_dvs/utils/dvs_util.py (hunk 2 above)
Comment (ekosareva) [inline vmware_dvs/utils/dvs_util.py:519]: All optimization logic containes in function that calls this one(please check vcenter_firewall._update_dvs_port_map).

File: vmware_dvs/agent/firewalls/vcenter_firewall.py (hunk 1 above)
Comment (VTabolin) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py:90]: I see that we use it on update rules and on port remove. Ok.

**Pull request merged**